source/Permissions.h

```diff
@@ -0,0 +1,31 @@
+/* Permissions.h
+Copyright (c) 2022 by Lemuria
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef ESKY_PERMISSIONS_H_
+#define ESKY_PERMISSIONS_H_
+
+#include "DataNode.h"
```

Comment:
Since `DataNode` only appears as a reference in a method argument, it can be forward declared. Include the full header in the .cpp file.
```suggestion
class DataNode;
```